### PR TITLE
Support multiple libPaths

### LINKED ISSUE #441: Support users with multiple lib paths
Another issue is `.libPaths()` in my case is a vector of 2 elements, therefore readLines couldn't parse the right index.css file.

_Originally posted by @xni7 in https://github.com/SafetyGraphics/safetyGraphics/pull/440#discussion_r528247859_

### PR DESCRIPTION
**Pull request by jwildfire**
# Overview

Iteratively looks for safetyCharts across multiple libPaths to fix #441. 

# Test Code

```
install.packages("safetyData")
library(safetyData)
devtools::install_github("safetyGraphics/safetyCharts", ref="dev")
library(safetyCharts)
devtools::install()
library(safetyGraphics)
safetyGraphics::safetyGraphicsApp()
```

### CODE REVIEW

File: R/makeChartConfig.R [hunk 1]
```diff
@@ -23,8 +23,10 @@
 makeChartConfig <- function(dirs, sourceFiles=TRUE){
     # Use the charts settings saved in safetycharts if no path is provided. 
     if(missing(dirs) || is.null(dirs)){
-        #dirs<-paste(.libPaths(),'safetycharts','chartSettings', sep="/")
-        dirs<-paste(.libPaths(),'safetycharts','config', sep="/")
+        for(lib in .libPaths()){
+            dirs<-paste(lib,'safetycharts','config', sep="/")
```
Comment (xni7) [inline R/makeChartConfig.R]: @jwildfire , should the path be `safetyCharts`, which is case sensitive on Macs and Linux?

**Comment by samussiah** [comment]
Hitting a few errors:
```
> safetyGraphicsApp()
Found 14 config files: aeExplorer, aeTimelines, hepexplorer, labdist, safetyHistogram, safetyOutlierExplorer, safetyOutlierExplorerModule, safetyOutlierExplorerStatic, safetyResultsOverTime, safetyResultsOverTimeStatic, safetyShiftPlot, tendril, tplyr_demog, tplyr_shift
Global Functions: aeExplorer_initlabdist_serverlabdist_uisafety_histogram_charttplyr_aes_charttplyr_shift_chart
+ aeExplorer: Found 1 of 1 workflow functions, and 0 other functions.
+ aeTimelines: Found 0 of 0 workflow functions, and 0 other functions.
+ hepexplorer: Found 0 of 0 workflow functions, and 0 other functions.
+ labdist: Found 2 of 2 workflow functions, and 0 other functions.
+ safetyHistogram: Found 0 of 0 workflow functions, and 0 other functions.
+ safetyOutlierExplorer: Found 1 of 1 workflow functions, and 0 other functions.
+ safetyOutlierExplorerModule: Found 2 of 2 workflow functions, and 0 other functions.
+ safetyOutlierExplorerStatic: Found 1 of 1 workflow functions, and 0 other functions.
+ safetyResultsOverTime: Found 1 of 1 workflow functions, and 0 other functions.
+ safetyResultsOverTimeStatic: Found 1 of 1 workflow functions, and 0 other functions.
+ safetyShiftPlot: Found 1 of 1 workflow functions, and 0 other functions.
+ tendril: Found 1 of 1 workflow functions, and 0 other functions.
+ tplyr_demog: Found 1 of 1 workflow functions, and 0 other functions.
+ tplyr_shift: Found 1 of 1 workflow functions, and 0 other functions.
Loading required package: shiny

Listening on http://127.0.0.1:4268
chartsTab() starting for aeExplorer
chartsTab() is initializing a widget at aeExplorer-wrap
chart is aeExplorer; package is safetyCharts
chartRenderWidget() starting for aeExplorer
chartsTab() starting for aeTimelines
chartsTab() is initializing a widget at aeTimelines-wrap
chart is aeTimelines; package is safetyCharts
chartRenderWidget() starting for aeTimelines
chartsTab() starting for hepexplorer
chartsTab() is initializing a widget at hepexplorer-wrap
chart is hepexplorer; package is safetyCharts
chartRenderWidget() starting for hepexplorer
chartsTab() starting for labdist
chartsTab() is initializing a module at labdist-wrap
chartsTab() starting for safetyHistogram
chartsTab() is initializing a widget at safetyHistogram-wrap
chart is safetyHistogram; package is safetyCharts
chartRenderWidget() starting for safetyHistogram
chartsTab() starting for safetyOutlierExplorer
chartsTab() is initializing a widget at safetyOutlierExplorer-wrap
chart is safetyOutlierExplorer; package is safetyCharts
chartRenderWidget() starting for safetyOutlierExplorer
chartsTab() starting for safetyOutlierExplorerModule
chartsTab() is initializing a module at safetyOutlierExplorerModule-wrap
chartsTab() starting for safetyOutlierExplorerStatic
chartsTab() starting for safetyResultsOverTime
chartsTab() is initializing a widget at safetyResultsOverTime-wrap
chart is safetyResultsOverTime; package is safetyCharts
chartRenderWidget() starting for safetyResultsOverTime
chartsTab() starting for safetyResultsOverTimeStatic
chartsTab() starting for safetyShiftPlot
chartsTab() is initializing a widget at safetyShiftPlot-wrap
chart is safetyShiftPlot; package is safetyCharts
chartRenderWidget() starting for safetyShiftPlot
chartsTab() starting for tendril
chartsTab() starting for tplyr_demog
chartsTab() starting for tplyr_shift
Warning: Error in : Only strings can be converted to symbols
  96: <Anonymous>
Warning: Error in : Only strings can be converted to symbols
  50: <Anonymous>
Warning: Error in : Only strings can be converted to symbols
  81: <Anonymous>
Warning: Error in : Only strings can be converted to symbols
  65: <Anonymous>
```

File: R/app_ui.R [hunk 2]
```diff
@@ -11,7 +11,13 @@
 
 app_ui <- function(meta, domainData, mapping, standards){
     #read css from pacakge
-    app_css <-  HTML(readLines( paste(.libPaths(),'safetyGraphics','safetyGraphics_app', 'www','index.css', sep="/")))
+    app_css <- NULL
+    for(lib in .libPaths()){
```
Comment (samussiah) [inline R/app_ui.R:15]: This seems to do the trick.

File: R/makeChartConfig.R [hunk 3]
```diff
@@ -23,8 +23,10 @@
 makeChartConfig <- function(dirs, sourceFiles=TRUE){
     # Use the charts settings saved in safetycharts if no path is provided. 
     if(missing(dirs) || is.null(dirs)){
-        #dirs<-paste(.libPaths(),'safetycharts','chartSettings', sep="/")
-        dirs<-paste(.libPaths(),'safetycharts','config', sep="/")
+        for(lib in .libPaths()){
+            dirs<-paste(lib,'safetyCharts','config', sep="/")
+            if(file.exists(dirs)) break
```
Comment (samussiah) [inline R/makeChartConfig.R:28]: So basically choose the first libPath.  This seems to do the trick.

**Review by samussiah — CHANGES_REQUESTED** [review]
The errors I hit previously have been fixed but I'm hitting new errors in a four anonymous functions: `Warning: Error in : Only strings can be converted to symbols`.  The full output is in a comment above.

**Comment by samussiah** [comment]
The `data.table(..., stringsAsFactors = TRUE)` idiosyncrasy popped up again in R/mod_mappingColumn.R.  Updating that setting fixed the app for me.  The only other issue I noticed is the Shift Table {Tplyr} + {kable} module was producing errors until I manually loaded Tplyr and kableExtra.

The _Participant Selector_ module will save me sooo much time so thank you all for that!  Please let me know how else I can help.

**Review by samussiah — APPROVED** [review]

File: R/makeChartConfig.R (hunk 3 above)
Comment (samussiah) [reply R/makeChartConfig.R:28]: Good point @xni7, it's not immediately obvious to me how `safetyCharts` fits into the workflow when using a custom set of chart config files.  It seems the workflow functions are pulled both from the location of the chart config files as well as from the `safetyCharts` package.

Comment (jwildfire) [reply R/makeChartConfig.R:28]: Yeah, you definitely *can* run `safetyGraphics` without `safetyCharts` if you supply all your own renderers and config files, but I'm not sure how often that will actually happen. 

For this chunk of code, `safetyCharts` *is* required unless the `dirs` option is set. So I think a warning is totally reasonable. I'll add something now and then merge.

**Comment by jwildfire** [comment]
Looks great! I added in @xni7 suggested warning (also created the related issues #517 for later) and cleaned up one failed test. Merging to dev now.

**Pull request merged**